pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/leap/LeapDriver.java

```diff
@@ -102,14 +109,19 @@ public String initialState() {
         log.trace("pollEvents: body={}", response.body());        
         JsonNode jsonNode = mapper.readTree(response.body());        
         final ArrayNode arrayNode = (ArrayNode) jsonNode;
+        Date maxTime = mapper.convertValue(arrayNode.get(0).get("receivedTimestamp"), Date.class);
```

Comment:
This will throw an exception if the array is empty. You should handle this case without throwing an exception.

Reply:
Done